File: DevExpress-Demo/VB/NavigationDemo.Wpf/ViewModels/FilterViewModel.vb
Imports System.Windows
Imports System.Windows.Media
Imports NavigationDemo.Utils

Namespace NavigationDemo

    Public Class FilterViewModel

        Private Shared filterPreviewImageSize As Size = New Size(200, 136)

        Public Sub New(ByVal uri As String, ByVal filter As FilterBase)
            Me.Filter = filter
            Me.Uri = uri
        End Sub

        Public Overridable Property Uri As String

        Public Overridable Property Image As ImageSource

        Public Overridable Property Filter As FilterBase

        Public Overridable ReadOnly Property Name As String
            Get
                Return If(Filter IsNot Nothing, Filter.Name, String.Empty)
            End Get
        End Property

        Public Overridable Sub OnUriChanged()
            Image = Filter.ApplyFilter(Uri, filterPreviewImageSize)
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/NavigationDemo.Wpf/ViewModels/FilterViewModel.cs
using System.Windows;
using System.Windows.Media;
using NavigationDemo.Utils;

namespace NavigationDemo {
    public class FilterViewModel {
        static Size filterPreviewImageSize = new Size(200, 136);

        public FilterViewModel(string uri, FilterBase filter) {
            Filter = filter;
            Uri = uri;
        }
        public virtual string Uri { get; set; }
        public virtual ImageSource Image { get; protected set; }
        public virtual FilterBase Filter { get; protected set; }
        public virtual string Name { get { return Filter != null ? Filter.Name : string.Empty; } }
        public virtual void OnUriChanged() {
            Image = Filter.ApplyFilter(Uri, filterPreviewImageSize);
        }
    }
}
